gcp_variant_transforms/libs/annotation/vep/vep_runner.py

```diff
@@ -0,0 +1,464 @@
+                # and this has to caught and communicated to the user.
+                'OTHER_VEP_OPTS':
+                    '--everything --check_ref --allow_non_variant',
+            },
+            'resources': {
+                'projectId': self._project,
+                'virtualMachine': {
+                    'disks': [
+                        {
+                            'name': 'vep',
+                            'sizeGb': _MINIMUM_DISK_SIZE_GB
+                        }
+                    ],
+                    'machineType': self._machine_type,
+                    # TODO(bashir2): Add the option of using preemptible
+                    # machines and the retry functionality.
+                    'preemptible': False,
+                    'serviceAccount': {
+                        'scopes': [
+                            'https://www.googleapis.com/auth/'
+                            'devstorage.read_write']
+                    }
+                },
+                'regions': [
+                    self._region
+                ]
+            }
+        }
+    }
+
+  def _make_action(self, *args, **kwargs):
+    # type: (str, str) -> Dict
+    command_args = list(args)
+    action = {
+        'commands': command_args,
+        'imageUri': self._vep_image_uri,
+        'mounts': [
+            {
+                'disk': 'vep',
+                'path': '/mnt/vep'
+            },
+        ]
+    }
+    action.update(kwargs)
+    # TODO(bashir2): Add a proper `label` based on command arguments.
+    return action
+
+  def _process_pipeline_args(self, pipeline_args):
+    # type: (List[str]) -> None
+    flags_dict = pipeline_options.PipelineOptions(
+        pipeline_args).get_all_options()
+    self._project = self._get_flag(flags_dict, 'project')
+    self._region = self._get_flag(flags_dict, 'region')
+    # TODO(bahsir2): Fix the error messages of _check_flag since
+    # --worker_machine_type has dest='machine_type'.
+    self._machine_type = self._get_flag(flags_dict, 'machine_type')
+    # TODO(bashir2): Fall back to num_workers if max_num_workers is not set.
+    self._max_num_workers = self._get_flag(flags_dict, 'max_num_workers')
+    if self._max_num_workers <= 0:
+      raise ValueError(
+          '--max_num_workers should be a positive number, got: {}'.format(
+              self._max_num_workers))
+
+  def _get_flag(self, pipeline_flags, flag):
+    # type: (Dict[str, Any], str) -> Any
+    if flag not in pipeline_flags or not pipeline_flags[flag]:
+      raise ValueError('Could not find {} among pipeline flags {}'.format(
+          flag, pipeline_flags))
+    logging.info('Using %s flag: %s.', flag, pipeline_flags[flag])
+    return pipeline_flags[flag]
+
+  def wait_until_done(self):
+    """Polls currently running operations and waits until all are done."""
+    while self._running_operation_ids:
+      self._running_operation_ids = [op for op in self._running_operation_ids
```

Comment:
ah, I thought you were checking for `done` operations over and over again and did not realize that `_running_operation_ids` is self-assigned. LGTM! :)